internal/server/apimigrator_test.go

```diff
@@ -33,21 +34,51 @@ func TestAddRequestRewrite(t *testing.T) {
 	})
 
 	get(a, router.Group("/"), "/test", func(c *gin.Context, req *upgradedTestRequest) (*api.EmptyResponse, error) {
-		assert.Assert(t, req.VegetableCount == 12)
+		assert.Equal(t, req.VegetableCount, 12)
 		return nil, nil
 	})
 
 	resp := httptest.NewRecorder()
 	req := httptest.NewRequest(http.MethodGet, "/test?cucumberCount=5&carrotCount=7", nil)
 	router.ServeHTTP(resp, req)
 
-	assert.Assert(t, resp.Result().StatusCode == 200)
+	assert.Equal(t, resp.Result().StatusCode, 200)
+}
+
+func TestStackedAddRequestRewrite(t *testing.T) {
+	srv := setupServer(t, withAdminIdentity)
+
+	a := &API{server: srv, disableOpenAPIGeneration: true}
+	router := gin.New()
+
+	addRequestRewrite(a, "get", "/test", "0.1.0", func(old legacyTestRequest) upgradedTestRequest {
+		return upgradedTestRequest{
+			VegetableCount: old.CarrotCount + old.CucumberCount,
+		}
+	})
+
+	addRequestRewrite(a, "get", "/test", "0.1.1", func(old upgradedTestRequest) upgradedTestRequest {
+		return upgradedTestRequest{
+			VegetableCount: old.VegetableCount * 2,
+		}
+	})
+
+	get(a, router.Group("/"), "/test", func(c *gin.Context, req *upgradedTestRequest) (*api.EmptyResponse, error) {
+		assert.Equal(t, req.VegetableCount, 24)
```

Comment:
might be clearer if this was something like `(5 + 7) * 2` so I don't need to do math